src/common/snippets/src/lowered/shape_inference/shape_inference.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "snippets/lowered/shape_inference/shape_inference.hpp"
+#include "snippets/lowered/shape_inference/shape_infer_instances.hpp"
+#include <openvino/op/util/unary_elementwise_arithmetic.hpp>
+#include <openvino/op/util/binary_elementwise_arithmetic.hpp>
+#include <openvino/op/util/binary_elementwise_comparison.hpp>
+#include <openvino/op/util/binary_elementwise_logical.hpp>
+#include <openvino/op/parameter.hpp>
+#include <openvino/op/result.hpp>
+#include <snippets/snippets_isa.hpp>
+
+namespace ov {
+namespace snippets {
+using ShapeInferPtr = IShapeInferSnippetsFactory::ShapeInferPtr;
+
+ShapeInferPtr IShapeInferSnippetsFactory::make(const ov::DiscreteTypeInfo& key, const std::shared_ptr<ov::Node>& op) {
+    const auto& maker_iter = registry.find(key);
+    if (maker_iter != registry.end())
+        return maker_iter->second(op);
+    return get_specific_op_shape_infer(key, op);
+}
+
+ShapeInferPtr IShapeInferSnippetsFactory::get_specific_op_shape_infer(const ov::DiscreteTypeInfo& key,
+                                                                      const std::shared_ptr<ov::Node>& op) const {
+    return {};
+}
+
+#define SHAPE_INFER_PREDEFINED(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>();} }
+#define SHAPE_INFER_OP_SPECIFIC(OP) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<OP::ShapeInfer>(n);} }
+#define SHAPE_INFER_OP_SPECIFIC_EXTERNAL(OP, InferType) \
+    { OP::get_type_info_static(), [](const std::shared_ptr<ov::Node>& n) { return std::make_shared<InferType>(n);} }
+
+const IShapeInferSnippetsFactory::TRegistry IShapeInferSnippetsFactory::registry {
```

Comment:
Could you please add the following ops as well? (They are elementwise ops but are derived explicitly from `Op`)
`ov::op::v1::LogicalNot`
`ov::op::v0::PRelu`
`ov::op::v1::Select`

Reply:
`LogicalNot` and `PRelu` added, `Select` was already there 🙂 
Thanks a lot!
BTW, why would they derive these ops from `Op` explicitly? It always a hassle with `PRelu` et al

Reply:
`ov::op::v1::LogicalNot` is unary binary elementwise operation. We don't have base common class for that 😢 
As for `PRelu` I don't remember to be honest, but we discussed it a few years ago. But I agree that it's a problematic op 😄